framework/auth/views.py

```diff
@@ -251,6 +287,62 @@ def confirm_email_get(token, auth=None, **kwargs):
     ))
 
 
+@collect_auth
+def verified_email_remove(auth=None):
+    """Called at login if user cancels their merge or email add.
+    methods: DELETE
+    """
+    user = auth.user
+    confirmed_email = request.get_json()
+    email_verifications = deepcopy(user.email_verifications)
+    for token in user.email_verifications:
+        if token == confirmed_email['token']:
```

Comment:
What if `token` isn't in the json body? A proper 400 should be returned to the client. You can do this validation before you go into the loop.

```python
json_body = request.get_json()
try:
    given_token = json_body['token']
except KeyError:
    raise HTTPError(...)
for token ...
```